src/main/webapp/app/exam/overview/exam-navigation-sidebar/exam-navigation-sidebar.component.ts

```diff
@@ -136,38 +137,38 @@ export class ExamNavigationSidebarComponent implements OnDestroy, OnInit {
     changePage(overviewPage: boolean, exerciseIndex: number, forceSave?: boolean, submission?: SubmissionVersion | ProgrammingSubmission | FileUploadSubmission): void {
         if (!overviewPage) {
             // out of index -> do nothing
-            if (exerciseIndex > this.exercises.length - 1 || exerciseIndex < 0) {
+            if (exerciseIndex > this.exercises().length - 1 || exerciseIndex < 0) {
                 return;
             }
             // set index and emit event
-            this.exerciseIndex = exerciseIndex;
+            this.exerciseIndex.update(() => exerciseIndex);
```

Comment:
Why are we using update and not set here? From my understanding update is used if we re-use the previous value 

```suggestion
            this.exerciseIndex.set(exerciseIndex);
```

Reply:
done